modules/build/src/main/scala/scala/build/preprocessing/ExtractedDirectives.scala

```diff
@@ -71,65 +51,21 @@ object ExtractedDirectives {
       else None
     if (malformedDirectiveErrors.isEmpty || maybeCompositeMalformedDirectiveError.isEmpty) {
 
-      def byKind(kind: UsingDirectiveKind) = all.find(_.getKind == kind).get
-
-      val codeDirectives           = byKind(UsingDirectiveKind.Code)
-      val specialCommentDirectives = byKind(UsingDirectiveKind.SpecialComment)
-      val plainCommentDirectives   = byKind(UsingDirectiveKind.PlainComment)
-
-      val directivesPositionsOpt =
-        if (
-          codeDirectives.containsTargetDirectivesOnly &&
-          specialCommentDirectives.containsTargetDirectivesOnly &&
-          plainCommentDirectives.containsTargetDirectivesOnly
-        )
+      val directives = allDirectives.head
```

Comment:
Are we sure that this list will always be non-empty?

Reply:
I think that was the assumption before and we never get an empty one, but we can add a check to be on the safe side.

Reply:
Ok changed to headOption, but now I realize that we could probably simplify the API and return a single `UsingDirective` instead of a list, but I don't want to release 2.0.0 of using directives just yet. There might some more things to fix.

In the meantime let's leave it as a list